javascript-source/systems/noticeSystem.js

```diff
@@ -120,18 +122,23 @@
                 rnd = Math.random(),
                 lastSent = lastTimeNoticesSent[idx],
                 time = (minTime + (maxTime - minTime) * rnd) * 6e4;
+
         if (isNaN(time)) {
             time = 10 * 6e4;
         }
         if (isNaN(lastSent)) {
             lastSent = now - time;
         }
         if (!retryCall && lastSent + time <= now) {
+            noticeLock.unlock();
+
```

Comment:
Locking multiple times from the same thread is safe on a ReentrantLock, as long as the same number of unlocks are performed

What I'm noticing is actually missing for safety is try-finally or try-catch-finally blocks to ensure there is no possibility of the unlock failing to trigger. The correct way to do this is:
```
lockobj.lock();
try {
    //Do Something
} catch (e) {  //Optional: only if Exceptions or JS errors are expected
    //Handle exception or JS error
} finally {
    lockobj.unlock();
}
```

The finally block is also triggered if a return statement occurs anywhere inside the try block

Reply:
Exactly which is why i'd like to cleanup the whole lock unlock mess and get rid of it.
I mostly just reordered the lock/unlock calls.

I'll scratch this PR and PR another bandaid for the initial issue. Another PR will follow with a cleaned up version of this if you approve with the mentioned rework since I dont see the point of adding hundreds of try chatches everywhere

Reply:
I don't know why I didn't catch this when the original PR was made for this system

Reply:
All good

The stop-gap-solution sounds seems good. I'll probably push another commit tomorrow with the requested changes.

A cleanup will follow soon(tm) then 😅

Reply:
I'm not necessarily opposed to using the lock, to ensure we can't manipulate the array from multiple threads at the same time, since each EventBus call is run on a random thread from the ThreadPool

Reply:
Still it doesnt seem necessary in other modules.
If thinking about threadsafety we'd need to rewrite a bunch of modules

Reply:
We probably do need to rewrite some LUL

Reply:
Thinking about concurenthashmaps or a synchronized collection which java provides should be usable here as well because of rhino.

That should reduce the need for manual locking/unlocking